### ISSUE TITLE
FR: Language argument support for the date filter

### ISSUE THREAD

**Issue by watarutmnh** [craftcms/cms#3006]
In some case I want to have the date in specific language format at site not same locale.
For example:
In Japanese site, if I write
```
entry.postDate|date('M j, Y')
```
it output in Japanese format like "6月 19, 2018".
but I want it to display "Jun 19, 2018" as in English format.

So tried...
```
{% set M = {'1月':'Jan', '2月':'Feb', '3月':'Mar', '4月':'Apr', '5月':'May', '6月':'Jun', '7月':'Jul', '8月':'Aug', '9月':'Sep', '10月':'Oct', '11月':'Nov', '12月':'Dec' } %}
{{ entry.postDate|date('M j, Y')|replace(M) }}
```

also suggested better solution by carlcs in #craft3-hlp at Slack.
```
{% set formatter = craft.app.i18n.getLocaleById('de').getFormatter %}
{{ formatter.asDatetime(date, 'php:M j, Y') }}
```
those work OK but It would be nice to have the `date` filter support for a language argument.

**Comment by AndrewMac** [comment]
Just to add to this, I have a site where we need the dates to come out in Thai, which has it's own set of numerals ( ๑๒๓๔๕๖๗๘๙๐ ). Not sure if that can also be covered in this request. 

To be honest, though, although dates are the most common need here, we did have another site where we needed to make sure that all numbers, even in body fields, ended up in Thai, so we had to run things through a macro. But that's not always essential. The dates-as-Thai-numbers is, though.

**Comment by AndrewMac** [comment]
Not quite the same issue, but we have a site where we need the dates to come out in Thai, which has it's own set of numerals ( ๑๒๓๔๕๖๗๘๙๐ ). And I'm sure there are other languages with their own locale-specific sets of numerals. Not sure if that can also be covered in this request.

To be honest, though, although dates are the most common need here, we did have another site where we needed to make sure that all numbers, even in body fields, ended up in Thai, so in the end we had to run most of our Twig output fields through a macro, which achieved what we were after with dates (and everything else).
